File: cache/src/main/java/com/mm/http/DefaultCacheConverterFactory.java
package com.mm.http;

import androidx.annotation.Nullable;


import com.mm.http.cache.CacheConverter;

import retrofit2.Response;

/**
 * Created by : majian
 * Date : 2021/9/29
 */

public class DefaultCacheConverterFactory extends CacheConverter.Factory {

    @Nullable
    @Override
    public CacheConverter<?> converterCache(RetrofitCache retrofit) {
        return new CacheConverter<Object>() {
            @Nullable
            @Override
            public Response<Object> convert(Response<Object> response) {
                return response;
            }
        };
    }

}

File: cache/src/main/java/com/mm/http/DefaultCacheConverterFactory.kt
package com.mm.http

import com.mm.http.cache.CacheConverter
import retrofit2.Response

/**
 * Created by : majian
 * Date : 2021/9/30
 */


class DefaultCacheConverterFactory : CacheConverter.Factory() {

    override fun <T> converterCache(retrofit: RetrofitCache): CacheConverter<T> {
        return object : CacheConverter<T> {
            override fun convert(response: Response<T>): Response<T>? {
                return response
            }
        }
    }
}